src/context/AirDBContext.tsx

```diff
@@ -71,6 +81,129 @@ export const AirDBProvider = React.memo((props: any) => {
     )
   }
 
+  const handleLogin = ({ password, username, setToggled }: any) => {
+    const isMatching =
+      users.filter(
+        (user: { fields: { password: string; username: string } }) =>
+          user.fields.password === password && user.fields.username === username
+      ).length > 0
+    const userId = (
+      users.filter(
+        (user: { fields: { username: string } }) =>
+          user.fields.username === username
+      )[0] || {
+        id: ''
+      }
+    ).id
+
+    if (isMatching) {
+      setToggled(false)
+      window.localStorage.setItem('username', username)
+      return updateAirDB('Testy - Users', userId, {
+        active: 'true'
+      })
+    } else {
+      throw new Error('INCORRECT PASSWORD')
+    }
+  }
+
+  const handleAddQuestion = ({ options, question }: any) => {
+    const userid = (
+      users.filter(
+        (user: { fields: { username: string } }) =>
+          user.fields.username === window.localStorage.getItem('username')
+      )[0] || { fields: { ID: '' } }
+    ).fields.ID
+
+    return postAirDB('Testy - Questions', {
+      ...question,
+      userid,
+      options: question.options
+        .sort()
+        .map((x: string) => x.trim())
+        .join(', '),
+      type: options.type,
+      autocheck: JSON.stringify(options.autocheck)
+    })
+  }
+
+  const handleAddTest = async ({ questionIDs, test }: any) => {
+    const userid = (
+      users.filter(
+        (user: { fields: { username: string } }) =>
+          user.fields.username === window.localStorage.getItem('username')
+      )[0] || { fields: { ID: '' } }
+    ).fields.ID
+
+    await postAirDB('Testy - Tests', {
+      ...test,
+      userid,
+      tags: test.tags.join(', '),
+      questions: questionIDs.join(', ')
+    }).then(() => (window.location.pathname = '/'))
+  }
+
+  const updateTestQuestions = ({ test, questionIDs }: any) => {
+    updateAirDB('Testy - Tests', test.id, {
+      questions: questionIDs.join(', ')
+    }).then(() => {
+      props.setRedirect('/')
+    })
+  }
+
+  const handleLogout = () => {
+    const userId = users.filter(
+      (user: any) =>
+        user.fields.username === window.localStorage.getItem('username')
+    )[0].id
+
+    return updateAirDB('Testy - Users', userId, {
+      active: false
+    })
+  }
+
+  const handleSetScore = ({ scoreID, score }: any) => {
+    const scorid = (
+      scores.filter(score => score.fields.ID === scoreID)[0] || {
+        id: ''
+      }
+    ).id
```

Comment:
Using `lodash`, this could be:
```javascript
import { find, get } from 'lodash'

const id = get(find(scores, score => score.fields.ID === scoreID), 'id', '');
```

Reply:
Is it alright if I do this in the most recent pull-request?

Reply:
Of course. 👍